### ISSUE TITLE
[Feature request] Bayesian Neural Networks as Inference Networks

### ISSUE THREAD

**Issue by philippreiser** [bayesflow-org/bayesflow#371]
We would like to use Bayesian Neural Networks (BNNs) as inference networks in BayesFlow, particularly for scenarios with very few training data points. In those settings standard neural networks tend to overfit while BNNs could provide uncertainty-aware inference for better generalization.

The paper [Low-Budget Simulation-Based Inference with Bayesian Neural Networks](https://arxiv.org/pdf/2408.15136) introduced this idea and there is an implementation available on [Github](https://github.com/ADelau/low_budget_sbi_with_bnn/tree/main).

Thank you!

**Comment by stefanradev93** [comment]
Very good idea!

**Comment by vpratz** [comment]
Sounds interesting.

Did anybody try to run the code from their repository and can comment on the computational requirements? In Appendix C, they state the "estimated computational cost is around 25, 000 GPU hours".

Also, they state in the conclusion: "A second limitation is the computational cost of predictions. When training a BNN using variational inference, the training cost remains on a similar scale as standard neural networks. At prediction time, however, the Bayesian model average described in Equation 4 must be approximated, and this requires a neural network evaluation for each Monte Carlo sample in the approximation. The computational cost of predictions then scales linearly with the number M of Monte Carlo samples."

If anyone tries it out or has a feeling what order of magnitude the inference time for a smaller benchmark might be on, I'd appreciate if you share an update here.

**Comment by stefanradev93** [comment]
Let's renew the discussion @LarsKue @paul-buerkner. Perhaps we can start by adding last layer BNNs via Laplace and Monte Carlo dropout for starters.

**Comment by paul-buerkner** [comment]
I think that could be a good start, if we think there are some relevant use-cases.
